corehq/elastic.py

```diff
@@ -48,7 +45,7 @@ def loads(self, s):
 
     def dumps(self, data):
         # don't serialize strings
-        if isinstance(data, six.string_types):
+        if isinstance(data, (str, bytes)):
```

Comment:
This results in a change of behaviour. Previously:
```python
  >>> ESJSONSerializer.dumps(b"foo")
  SerializationError: Object of type 'bytes' is not JSON serializable
```
Now:
```python
  >>> ESJSONSerializer.dumps(b"foo")
  b'foo'
```

If the caller is expecting a string and gets bytes instead, that breaks the Unicode sandwich, and could result in harder-to-debug downstream errors.